fendermint/abci/src/application.rs

```diff
@@ -62,20 +64,12 @@ pub trait Application {
         &self,
         request: request::PrepareProposal,
     ) -> AbciResult<response::PrepareProposal> {
-        let max_tx_bytes: usize = request.max_tx_bytes.try_into().unwrap();
-        let mut size: usize = 0;
-        let mut txs = Vec::new();
-        for tx in request.txs {
-            if size.saturating_add(tx.len()) > max_tx_bytes {
-                break;
-            }
-            size += tx.len();
-            txs.push(tx);
-        }
+        let txs = take_until_max_size(request.txs, request.max_tx_bytes.try_into().unwrap());
```

Comment:
We only check here if we reach the size limit of the Tendermint block, right? Should we add also here a `TODO` to remember not to exhaust the gas limit of a block?

Reply:
The gas aspect is definitely ignored at the moment, thanks for the reminder. I created an issue to track it: https://github.com/consensus-shipyard/fendermint/issues/208

The check would not be here because at this size we don't know what messages we are dealing with, but once they are parsed into `ChainMessage` we can look at the gas limits.